addons/portal/controllers/portal.py

```diff
@@ -216,7 +216,7 @@ def account(self, redirect=None, **post):
         states = request.env['res.country.state'].sudo().search([])
 
         values.update({
-            'partner': partner,
+            'partner_sudo': partner,
```

Comment:
You should remove the old one in stable. The original view at the time of the freeze needs to still work as people don't have to update their views, and any external override too.

What's more partner isn't even sudo here, so why name it like that?